awx/sso/backends.py

```diff
@@ -423,3 +468,52 @@ def on_populate_user(sender, **kwargs):
     if profile.ldap_dn != ldap_user.dn:
         profile.ldap_dn = ldap_user.dn
         profile.save()
+
+    reconsile_users_org_team_mappings(user, desired_org_states, desired_team_states, 'LDAP')
+
+
+def reconsile_users_org_team_mappings(user, desired_org_states, desired_team_states, source):
+    from awx.main.models import Organization, Team
+
+    org_content_type = ContentType.objects.get_for_model(Organization)
+    team_content_type = ContentType.objects.get_for_model(Team)
+
+    # users_roles is a flat list of IDs
+    users_roles = list(user.roles.filter(content_type__in=[org_content_type, team_content_type]).values_list('pk', flat=True))
```

Comment:
Would be technically faster as `set`, and it looks like the code allows for that.